src/overlays/actors/ovl_Bg_Crace_Movebg/z_bg_crace_movebg.c

```diff
@@ -183,117 +205,145 @@ void BgCraceMovebg_Update(Actor* thisx, PlayState* play) {
     BgCraceMovebg* this = THIS;
     s32 pad;
     Player* player = GET_PLAYER(play);
-    s16 tempRot;
+    s16 yawDiff;
 
-    switch (BGCRACEMOVEBG_GET_F(&this->dyna.actor)) {
-        case 0:
-        case 2:
-            tempRot = this->dyna.actor.yawTowardsPlayer - this->dyna.actor.home.rot.y;
-            if ((tempRot >= -0x4000) && (tempRot <= 0x4000)) {
+    switch (BG_CRACE_MOVEBG_GET_TYPE(&this->dyna.actor)) {
+        case BG_CRACE_MOVEBG_TYPE_CLOSING:
+        case BG_CRACE_MOVEBG_TYPE_UNUSED_CLOSING:
+            yawDiff = this->dyna.actor.yawTowardsPlayer - this->dyna.actor.home.rot.y;
+            if ((yawDiff >= -0x4000) && (yawDiff <= 0x4000)) {
                 this->dyna.actor.shape.rot.y = this->dyna.actor.home.rot.y;
             } else {
                 this->dyna.actor.shape.rot.y = this->dyna.actor.home.rot.y + 0x8000;
             }
+            // Fallthrough
+
         default:
             this->actionFunc(this, play);
-            Math_Vec3f_Copy(&this->unk188, &player->bodyPartsPos[0]);
+            Math_Vec3f_Copy(&this->prevPlayerPos, &player->bodyPartsPos[0]);
```

Comment:
Does this even have to be in the switch if everything does it?

Reply:
No, it does not need to be in the switch.